pkg/gather/collect/collect.go

```diff
@@ -357,3 +228,48 @@ func (c *Collector) CollectResources(ctx context.Context, resourceInfo *Resource
 
 	return apimachineryutilerrors.NewAggregate(errs)
 }
+
+func (c *Collector) CollectAllResources(ctx context.Context, resources []*ResourceInfo) error {
```

Comment:
Does the name `CollectAllResources` still make sense?

How about `CollectAllDiscoveredResources` ?